### ISSUE TITLE
Type support for captured clients and aws-sdk (without depending on aws-sdk)

### ISSUE THREAD

**Issue by LosD** [aws/aws-xray-sdk-node#276]
There's a major drawback when using Typescript or type completion with alpha.2: For some reason, capturing `captureAWSClient()` does not return the same type that went in, but `any`. That seems wrong.

A better function definition that doesn't depend on aws-sdk, would to me be something like:
```typescript
export function captureAWS<T>(awssdk: T): T;

export function captureAWSClient<T>(service: T): T;
```

Without something like that, the patched services/SDK is suddenly a bit of a hassle to use.

IMHO, using `any` for... well, anything (no pun intended) in a library should not go through review without a _very_ good reason, especially for outputs.

**Comment by willarmiros** [comment]
@LosD thank you very much for this suggestion! My apologies for being newer to TypeScript conventions. I'll be sure to include it in a PR before v3 is released and remove the guidance about using type assertions for the `captureAWS` APIs.

**Comment by LosD** [comment]
Sounds good!

Not sure if it really a convention, it's just that it's not hugely practical or safe to get an output without any hints for type checking and IDEs.

(I must admit to having missed the new guidance, I just upgraded, saw the `any`, and thought that was a bit heavy handed way to treat the poor types 😉)

**Comment by willarmiros** [comment]
Released in 3.0.0.
